mailet/standard/src/main/java/org/apache/james/transport/mailets/ContactExtractor.java

```diff
@@ -95,26 +101,35 @@ public void service(Mail mail) throws MessagingException {
         }
     }
 
-    private Optional<String> extractContacts(Mail mail) throws MessagingException, IOException {
-        MimeMessage message = mail.getMessage();
+    @VisibleForTesting
+    Optional<String> extractContacts(Mail mail) throws MessagingException, IOException {
+        ImmutableList<String> allRecipients = getAllRecipients(mail.getMessage());
 
         return Optional.of(mail.getSender())
             .map(MailAddress::asString)
-            .filter(Throwing.predicate(sender -> hasRecipients(message)))
-            .map(Throwing.function(sender -> new ExtractedContacts(sender, recipients(message))))
+            .filter(sender -> !allRecipients.isEmpty())
+            .map(sender -> new ExtractedContacts(sender, allRecipients))
             .map(Throwing.function(extractedContacts -> objectMapper.writeValueAsString(extractedContacts)));
     }
 
-    @VisibleForTesting boolean hasRecipients(MimeMessage mimeMessage) throws MessagingException {
-        return mimeMessage.getAllRecipients() != null 
-                && mimeMessage.getAllRecipients().length > 0;
+    private ImmutableList<String> getAllRecipients(MimeMessage mimeMessage) throws MessagingException {
+        return StreamUtils.flatten(
```

Comment:
why do you to wrap getRecipients in a Stream ?

Reply:
because it's more easy to concatenate using flatten

Reply:
We lack the vararg version of `flatten` then. `Stream.of` is simpler because it provides vararg variant.

Reply:
:+1: